docs/appendices/release-notes/unreleased.rst

```diff
@@ -243,3 +243,11 @@ Fixes
     UNION
     SELECT i, avg(k), j FROM t3 GROUP BY 1, 3
     LIMIT 10
+
+- Fixed an issue which prevented ``INSERT INTO ... SELECT ...`` from inserting
+  any records if the target table had a generated column referring to a
+  partitioned column of a none-string type.
+
+- Fixed an issue which prevented ``INSERT INTO ... SELECT ...`` from inserting
+  any records if the target table had a partitioned column with a ``CHECK``
+  constraint on a column of a none-string type.
```

Comment:
As discussed, still need to move validation to the earlier phase. 

Adding a change entry already here for the CHECK issue since it's a fix from the user perspective.

I will create a follow up PR to fix last remaining PARTITIONED BY related bugs which is about generated expression validation. The follow up PR will also include moving validation to the earlier phase so that partitions are not left behind in case of validation failures.